Add documentation page using mkdocs

SarahAlidoost opened the pull request:
closes #2 

The github workflow is run when this PR is merged with `main` branch. 

install dependencies:
```
pip install mkdocs
pip install mkdocs-material
pip install mkdocs-jupyter
pip install mkdocstrings[python]
pip install mkdocs-gen-files
pip install mkdocs-include-markdown-plugin
```

preview:
```
mkdocs serve
```

TODO:
- update readme of all repos  #7
- add a contributing file to this repo #6
- add a citation file #3
- explore a workflow for running GA when a readme is updated.

fnattino commented on docs/components/jupyter_dask_cloud_to_cluster.md:
```diff
@@ -0,0 +1,7 @@
+#
```
This top "#" character blocks the titles from the original repository to be displayed - If I remove it, the title from the repository README.md  (e.g. "JupyterDask on SLURM") is properly displayed. 

Or is there some other reason why this should be here?

```suggestion

```

SarahAlidoost replied:
The reason is that the "title" is displayed in the left panel. If `#` is removed, the tile is again shown on the top of the page.  I think displaying the title two times makes the page look busy, see the screenshot below. WHat do you think? :thinking: 
![Screenshot from 2024-06-28 15-11-10](https://github.com/RS-DAT/RS-DAT/assets/55081872/3b40b3df-4fcd-4a7a-b674-9f35ac51d7a2)

fnattino replied:
Oh, I actually like it with the title on both sides, it looks clearer to me which tab I am actually looking at. But if you like it better without the title, feel free to remove it :)

fnattino commented on docs/components/jupyter_dask_on_slurm.md:
```diff
@@ -0,0 +1,7 @@
+#
```
See comment above:

```suggestion

```

fnattino commented on docs/components/jupyter_dask_on_src.md:
```diff
@@ -0,0 +1,8 @@
+#
```
See comment above:

```suggestion

```

fnattino commented on docs/components/jupyter_docker_spawner_on_src.md:
```diff
@@ -0,0 +1,7 @@
+#
```
See comment above:

```suggestion

```

fnattino commented on docs/components/jupyterdask_with_singularity.md:
```diff
@@ -0,0 +1,7 @@
+#
```
See comment above:

```suggestion

```

fnattino commented on docs/components/jupyter_dask_cloud_to_cluster.md:
Maybe the folder could be "deploying" rather than "components"? Ultimately, all of these are different ways of deploying RS-DAT rather than pieces of it. (I would also avoid "deployments" since this sounds more like "platforms that you can access"). What do you think?

SarahAlidoost replied:
good point, it is renamed to "deploying"

fnattino commented on docs/components/jupyter_dask_cloud_to_cluster.md:
```diff
@@ -0,0 +1,7 @@
+#
+
+{% include-markdown "https://raw.githubusercontent.com/RS-DAT/JupyterDaskCloud2Cluster/main/README.md" start="# JupyterDaskCloud2Cluster" heading-offset=2 %}
+
+## Source code of the tool
+
+The source code of this tool can be found at [RS-DAT/JupyterDaskCloud2Cluster](https://github.com/RS-DAT/JupyterDaskCloud2Cluster).
```
I love the idea of including the README from the various repositories and link to the GH repos for more information! I would however put the links to the repo at the top, so that this is immediately visible. Also maybe we can use the GH badge in the following way? 

```suggestion

GitHub repository: [![](https://img.shields.io/badge/github-repo-000.svg?logo=github&labelColor=gray&color=blue)][link]

{% include-markdown "https://raw.githubusercontent.com/RS-DAT/JupyterDaskCloud2Cluster/main/README.md" start="# JupyterDaskCloud2Cluster" heading-offset=2 %}

[link]: https://github.com/RS-DAT/JupyterDaskCloud2Cluster
```

fnattino commented on docs/demos/2022_12_01_demo_remote_sensing_session.md:
Great, we finally have on overview of all demos! :) 

I have the same comment as for the deployment instructions, I would add the link to the repository to the top.

SarahAlidoost replied:
it is added now.

fnattino commented on docs/demos/2022_12_01_demo_remote_sensing_session.md:
```diff
@@ -0,0 +1,7 @@
+
+## Source code
+
+The source code of this tool can be found at [RS-DAT/2022-12-01-demo-remote-sensing-session](https://github.com/RS-DAT/2022-12-01-demo-remote-sensing-session).
```
```suggestion
The source code of the material can be found at [RS-DAT/2022-12-01-demo-remote-sensing-session](https://github.com/RS-DAT/2022-12-01-demo-remote-sensing-session).
```

fnattino commented on docs/examples.md:
Amazing way to collect all examples!

fnattino commented on mkdocs.yml:
```diff
@@ -0,0 +1,74 @@
+repo_url: https://github.com/RS-DAT/RS-DAT
+repo_name: RS-DAT
+
+nav:
+  - What can RS-DAT do for you: index.md
+  - Deployable Analysis Tools:
+    - Jupyter Dask On SLURM: components/jupyter_dask_on_slurm.md
+    - Jupyter Dask On SRC: components/jupyter_dask_on_src.md
+    - Jupyter Dask Cloud to Cluster: components/jupyter_dask_cloud_to_cluster.md
+    - Jupyter Docker Spawner On SRC: components/jupyter_docker_spawner_on_src.md
+    - Jupyter Dask with Singularity: components/jupyterdask_with_singularity.md
+  - Examples: examples.md
+  - Demos:
+    - 2023-05-31-SummerSchool-HDCRS: demos/2023_05_31_SummerSchool_HDCRS.md
+    - 023-05-22-scalable-GIS: demos/023_05_22_scalable_GIS.md
+    - 2023-05-15-workshop-ITC: demos/2023_05_15_workshop_ITC.md
+    - 2023-03-10-demo-TU-Delft: demos/2023_03_10_demo_TU_Delft.md
+    - 2022-12-01-demo-remote-sensing-session: demos/2022_12_01_demo_remote_sensing_session.md
+  - How to contribute: how_to_contribute.md
+  - How to cite: how_to_cite.md
+  - License: what_license.md
```
The top bar seems quite busy - what do you think of simplifying some of the headers?

```suggestion
nav:
  - Introduction: index.md
  - Deploy:
    - Jupyter Dask On SLURM: components/jupyter_dask_on_slurm.md
    - Jupyter Dask On SRC: components/jupyter_dask_on_src.md
    - Jupyter Dask Cloud to Cluster: components/jupyter_dask_cloud_to_cluster.md
    - Jupyter Docker Spawner On SRC: components/jupyter_docker_spawner_on_src.md
    - Jupyter Dask with Singularity: components/jupyterdask_with_singularity.md
  - Examples: examples.md
  - Demos:
    - 2023-05-31-SummerSchool-HDCRS: demos/2023_05_31_SummerSchool_HDCRS.md
    - 023-05-22-scalable-GIS: demos/023_05_22_scalable_GIS.md
    - 2023-05-15-workshop-ITC: demos/2023_05_15_workshop_ITC.md
    - 2023-03-10-demo-TU-Delft: demos/2023_03_10_demo_TU_Delft.md
    - 2022-12-01-demo-remote-sensing-session: demos/2022_12_01_demo_remote_sensing_session.md
  - Contributing: how_to_contribute.md
  - Cite: how_to_cite.md
  - License: what_license.md
```

fnattino reviewed (commented):
Amazing work @SarahAlidoost, I love it! I have added few comments, mainly style things, feel free to include only what you agree on. Thanks a lot.

SarahAlidoost commented on .github/workflows/doc_deploy.yml:
```diff
@@ -0,0 +1,31 @@
+          python-version: "3.11"
+      - name: Install dependencies
+        run: |
+          python -m pip install mkdocs
```
good point, it is added.

SarahAlidoost commented:
> Amazing work @SarahAlidoost, I love it! I have added few comments, mainly style things, feel free to include only what you agree on. Thanks a lot.

@fnattino thanks for reviewing and helpful suggestions. I really liked the idea of adding github badges :+1:  I addressed your comments except one about the top `#` in the readme files, see my answer [here](https://github.com/RS-DAT/RS-DAT/pull/5#discussion_r1658726773). 

I also used the mkdocs extension `admonition` to change the look, see screenshots below. If something needs to be changed, please let me know :blush: 

here **Deploy**:
![deploy_screenshot](https://github.com/RS-DAT/RS-DAT/assets/55081872/5f19b07c-8c68-4c38-b7e5-f3cdaf522d61)

and here **Demos**:
![demos](https://github.com/RS-DAT/RS-DAT/assets/55081872/6c53919f-524e-49d9-adae-b19a6d6de331)

and here **Examples**:

![ML_screenshot](https://github.com/RS-DAT/RS-DAT/assets/55081872/ebce9a52-a9d0-41d3-8058-1382462fdccc)

fnattino reviewed (approved):
Super, @SarahAlidoost, looks great! Even nicer with the new style option!
